packages/block-editor/src/components/inserter/index.js

```diff
@@ -56,10 +64,17 @@ class Inserter extends Component {
 	renderToggle( { onToggle, isOpen } ) {
 		const {
 			disabled,
+			hasOnlyOneAllowedInserterItem,
+			blockTitle,
+			insertTheOnlyAllowedItem,
 			renderToggle = defaultRenderToggle,
 		} = this.props;
 
-		return renderToggle( { onToggle, isOpen, disabled } );
+		if ( hasOnlyOneAllowedInserterItem ) {
```

Comment:
Could probably make this simpler and avoid rendering the dropdown completely if there's on a single block. I'm thinking an early return that just renders the `IconButton`.

Reply:
Something like:
```
render() {
if ( hasOnlyOneAllowedBlock ) {
  return this.renderToggle( { onToggle: insertTheOnlyAllowedItem } );
}
```